### PR TITLE
root dir for execdriver (/var/run/docker) should be configurable

### LINKED ISSUE #13082: root dir for execdriver (/var/run/docker) should be configurable
Unlike docker runtime dir, which can be configured with ```--graph`` option,
there doesn't seem to be any way to manually set execdriver's rootdir.

Seems the hard-coded ```/var/run/docker``` path was introduced with this PR #11716

It'd be great if we could configure this with command line arg?

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan shishir.mahajan@redhat.com

fixes #13082

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
Fixes #13082

**Comment by jfrazelle** [comment]
I think the real question is why? Why would someone ever need to change these files. It's playing with fire.

**Comment by tianon** [comment]
This would make the proper fix for #10567 _much_ simpler, so I'm personally +1.

**Comment by tianon** [comment]
(or at least half that fix)

**Comment by calavera** [comment]
it sounds reasonable and consistent with other configuration flags, I guess it's up to you if you want to play with :fire:

**Comment by jfrazelle** [comment]
i see the usecase for #10567

**Comment by crosbymichael** [comment]
code LGTM

this should have docs

**Comment by vieux** [comment]
code LGTM

**Comment by shishir-a412ed** [comment]
@crosbymichael I have updated the man page doc. Let me know if it needs to be updated at any other places.

**Comment by crosbymichael** [comment]
ping @moxiegirl

**Comment by thaJeztah** [comment]
Needs some docs in https://github.com/docker/docker/blob/master/docs/sources/reference/commandline/cli.md#daemon and probably below "Options for the native execdriver" (?)

Should this be a separate option though? Or passed as option to `--exec-opt`?

**Comment by shishir-a412ed** [comment]
@thaJeztah Will update the docs. Not related to this PR, but exec-opt[] seems to be missing from the docs. Opened a quick PR for that.

https://github.com/docker/docker/pull/13127

**Comment by thaJeztah** [comment]
> exec-opt[] seems to be missing from the docs. Opened a quick PR for that.

Good call! Thnx

File: docs/man/docker.1.md [hunk 1]
```diff
@@ -56,6 +56,9 @@ To see the man page for a command run **man docker <command>**.
 **--exec-opt**=[]
   Set exec driver options. See EXEC DRIVER OPTIONS.
 
+**--exec-root**=""
+  Path to use as the root of the Docker execdriver. Default is `/var/run/docker`.
+
```
Comment (thaJeztah) [inline docs/man/docker.1.md]: Is this option only available for the native driver, or also for LXC? That could be important information to add (also in the other suggested parts of the documentation)

**Comment by thaJeztah** [comment]
@crosbymichael @tianon @calavera @jfrazelle any thoughts on my earlier remark;

> Should this be a separate option though? Or passed as option to `--exec-opt`?

Sorry for suggesting after design-review has passed, just wondered if we want to add a dedicated flag/option for a (probably) not-so-frequently used option.

**Comment by jfrazelle** [comment]
it effects both... so idk

**Comment by jfrazelle** [comment]
we already did design review...

**Comment by jfrazelle** [comment]
we have graph-dir and that is not  in the storage opts

**Comment by thaJeztah** [comment]
> we have graph-dir and that is not in the storage opts

yes :'( I noticed. do we want to add more flags? thought I should (at least) mention because, idk, now there's still a chance to decide otherwise. (You must all now starting to hate me :))

**Comment by jfrazelle** [comment]
idk I think its fine as is, but other people can chime in

**Comment by jfrazelle** [comment]
its in the master docs

**Comment by shishir-a412ed** [comment]
@jfrazelle docs and bash completion updated.
ping @crosbymichael

**Comment by moxiegirl** [comment]
LGTM

**Comment by moxiegirl** [comment]
ping @thaJeztah no response on your LXC issue, perhaps you are willing to LGTM as is and we can open another PR if you are right.

File: docs/man/docker.1.md (hunk 1 above)
Comment (calavera) [inline docs/man/docker.1.md]: this option is for any execdriver, afaik.

**Comment by thaJeztah** [comment]
@moxiegirl bit late in my response, was watching the Niagara Falls, :smile:. It was just answered in https://github.com/docker/docker/pull/13086#discussion_r30374416.

not entirely sure if this shouldn't be added as `--exec-opt` but won't block this from being merged.

LGTM

**Comment by thaJeztah** [comment]
sorry, @shishir-a412ed, looks like this needs a rebase :'(

**Comment by shishir-a412ed** [comment]
@thaJeztah rebased.

**Comment by calavera** [comment]
LGTM, merging! :fireworks:

**Pull request merged**